File: solutions/javascript/raindrops/1/raindrops.js
export const convert = (n) => {
    const sounds = new Map([
        [3, "Pling"],
        [5, "Plang"],
        [7, "Plong"],
    ]);

    let result = "";
    for (const [k, v] of sounds) if (n % k === 0) result += v;
    return result || String(n);
};

File: solutions/python/raindrops/1/raindrops.py
def convert(number: int):
    sounds = [
        "Pling" if number % 3 == 0 else "",
        "Plang" if number % 5 == 0 else "",
        "Plong" if number % 7 == 0 else "",
    ]

    match "".join(sounds):
        case "":
            return str(number)
        case sound:
            return sound
